### PR TITLE
Feature/access project from job

### LINKED ISSUE #423: Accessing Project from Job
### Feature description

Sometimes I need to access the `Project` object from a `job`, for example to access the project-wide `data` attribute. Currently, in order to achieve this, a module-wide variable has to be created for the project, but this could span out over several modules and need to load/import the project from each one.

### Proposed solution

Exposing the already present `job._project` as `job.project`. If this is not possible in the near future, the current way of doing things could be explained in the docs (I was not able to find it, if it is not there I could add it myself).

### PR DESCRIPTION
**Pull request by AlainKadar**
## Description
Users can access the project which a job belongs to from the job object

## Motivation and Context
Resolves #423 

## Checklist:

- [x] I am familiar with the [Contributing Guidelines](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [Contributor Agreement](https://github.com/glotzerlab/signac/blob/master/ContributorAgreement.md).
- [ ] My name is on the [list of contributors](https://github.com/glotzerlab/signac/blob/master/contributors.yaml).
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.
- [x] The [package documentation](https://github.com/glotzerlab/signac/tree/master/doc) and [framework documentation](https://docs.signac.io/) in [signac-docs](https://github.com/glotzerlab/signac-docs) are up to date with these changes.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/signac/blob/master/changelog.txt) and added any related issue and pull request numbers for future reference.

### CODE REVIEW

File: signac/contrib/job.py [hunk 1]
```diff
@@ -606,6 +606,15 @@ def data(self, new_data):
...
 
+    @property
+    def project(self):
+        """Return the project that contains this job."""
+        return self._project
+
+    @project.setter
+    def project(self, new_project):
+        raise ValueError("Setting project attribute directly is not allowed")
+
```
Comment (bdice) [inline signac/contrib/job.py:617]: Just delete this -- without a `setter`, the property will not be settable and will raise an `AttributeError`.

```suggestion
```

File: tests/test_job.py [hunk 2]
```diff
@@ -238,6 +238,13 @@ def test_deepcopy(self):
         copied_job.sp.a = 3
         assert copied_job in self.project
 
+    def test_project(self):
+        job = self.project.open_job({"a": 0})
+        assert job.project.path == self._tmp_pr
```
Comment (bdice) [inline tests/test_job.py:243]: Let's test a few more invariants.
```suggestion
        assert isinstance(job.project, signac.Project)
        assert job in job.project
        assert job.project.path == self._tmp_pr
```

**Review by bdice — CHANGES_REQUESTED** [review]
@AlainKadar Can you rebase this on the latest `next` branch? It looks like there are some extra commits on this branch.

**Comment by AlainKadar** [comment]
Sorry, I completely messed up my local repo with changes from master, so moved this to a new PR: #808

**Pull request closed without merging**